src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -57,9 +57,18 @@ static PSVersionInfo()
         {
             s_psVersionTable = new PSVersionHashTable(StringComparer.OrdinalIgnoreCase);
 
+            string assemblyPath = typeof(PSVersionInfo).GetTypeInfo().Assembly.Location;
+            FileVersionInfo fileVersionInfo = FileVersionInfo.GetVersionInfo(assemblyPath);
+
+            // Convert to '6.0.0-beta.7-29-g52c6bfe1eae24dbaa1a162bffb3754ba3fdc1f4c'.
+            // The raw git commit version string is widely used internally.
```

Comment:
Please add comments about the two formats of the product version strings. For example:
```
// Get ProductVersion of the assembly. The product version string can be one of the following format examples:
//    when powershell is built from a commit -- '6.0.0-beta.7 Commits: 29 SHA: 52c6b...'
//    when powershell is built from a release tag -- '6.0.0-beta.7 SHA: f1ec9...'
```
> // Convert to '6.0.0-beta.7-29-g52c6bfe1eae24dbaa1a162bffb3754ba3fdc1f4c'.

This is not accurate. When building from a release tag, it should be converted to `v6.0.0-beta.7`.

Reply:
Sorry - I lost the commit.

Fixed.